### PR TITLE
Add cisco IMC MIBs

### PR DESCRIPTION
**Pull request by ranmalka**
Cisco IMC MIBS to monitor Cisco CUCS server IMC (parallel to HPE-ILO)

@SuperQ 
@bastischubert 
@RichiH 
Signed-off-by: ran rmalka@trigoretail.com

### CODE REVIEW

**Comment by ranmalka** [comment]
@bastischubert I reduced it to 3 metrics instead of 3000 lines, hope its ok now

File: snmp.yml [hunk 1]
```diff
@@ -2962,7 +3029,6 @@ modules:
       indexes:
       - labelname: bsnAPDot3MacAddress
         type: PhysAddress48
-        fixed_size: 6
```
Comment (bastischubert) [inline snmp.yml:2965]: this should not have happend - do you remember how you ran the generator and which version it was?

Comment (ranmalka) [reply snmp.yml:2965]: added the mibs files to the generator/mibs

make generator mibs
make generate

./generator generate

copy the generator/snmp.yml to ../snmp.yml

File: generator/Makefile [hunk 2]
```diff
@@ -399,3 +400,10 @@ $(MIBDIR)/readydataos:
 $(MIBDIR)/powercom:
 	@echo ">> Downloading powercom"
 	@curl $(CURL_OPTS) -o $(MIBDIR)/powercom "$(POWERCOM_URL)"
+
+$(MIBDIR)/cisco_imc:
```
Comment (bastischubert) [inline generator/Makefile:404]: this target needs to be referenced near line 159 to actually be called during `make mibs`.

File: generator/Makefile [hunk 3]
```diff
@@ -399,3 +400,10 @@ $(MIBDIR)/readydataos:
...
+	$(eval TMP := $(shell mktemp))
+	$(eval TMP_DIR := $(shell mktemp -d))
+	@echo ">> Downloading Cisco UCS to $(TMP)"
+	@curl $(CURL_OPTS) $(CURL_USER_AGENT) -o $(TMP) $(CISCO_CUCS_URL)
+	@tar -xf $(TMP) -C $(MIBDIR)/cisco_imc
```
Comment (bastischubert) [inline generator/Makefile:409]: this dir `$(MIBDIR)/cisco_imc` does not exist, we'd either need to create it or even better just extract the two used mibfiles (`CISCO-UNIFIED-COMPUTING-FAULT-MIB.my` `CISCO-UNIFIED-COMPUTING-STORAGE-MIB.my`)

Comment (ranmalka) [reply generator/Makefile:410]: I see that CISCO-UNIFIED-COMPUTING-FAULT-MIB.my CISCO-UNIFIED-COMPUTING-STORAGE-MIB.my, they have many dependencies like CISCO-UNIFIED-COMPUTING-TC-MIB; CISCO-UNIFIED-COMPUTING-MIB CISCO-SMI CISCO-TC
that I know I can found that in the tar file.
What do you think I should do?

Comment (ranmalka) [reply generator/Makefile:411]: fixed that with line 408, wdyt? @bastischubert

**Comment by ranmalka** [comment]
@bastischubert still `make generate` failed here but worked locally, how do I fix that?

`time=2024-11-27T11:36:46.700Z level=INFO source=main.go:57 msg="Generating config for module" module=cisco_imc
time=2024-11-27T11:36:46.733Z level=ERROR source=main.go:138 msg="Error generating config netsnmp" err="cannot find oid '1.3.6.1.4.1.9.9.719.1.1.1.1.11' to walk"
make: *** [Makefile:95: generate] Error 1`

File: snmp.yml (hunk 1 above)
Comment (SuperQ) [reply snmp.yml]: This generally indicates a miss-match problem with net-snmp. We should have fixed issue was fixed a while ago with some CGO workarounds. However, I think some MacOS users were still seeing issues.

**Comment by bastischubert** [comment]
@ranmalka have you checked my last comments? after that i think we're ready to merge

**Comment by ranmalka** [comment]
> @ranmalka have you checked my last comments? after that i think we're ready to merge

@bastischubert do you mean to put the files directly in the mibs location? so yes that's what I did

**Comment by bastischubert** [comment]
@ranmalka i left a few comments that still need to be adressed with the `Makefile`

**Comment by ranmalka** [comment]
@bastischubert conflicts have been fixed :)

**Comment by bastischubert** [comment]
@ranmalka did you notice the comments i left and are still not yet addressed ? https://github.com/prometheus/snmp_exporter/pull/1293/files you should see them there.

- i asked why you added the CISCO_FRU_URL which seems unused.
- `$(MIBDIR)/cisco_imc` to `$(MIBDIR)/.cisco_imc` still needs to be addressed

everything else already looks good and ready to get merged 👍

**Comment by ranmalka** [comment]
@bastischubert CISCO_FRU_URL used by 1.3.6.1.4.1.9.9.719.1.1.1.1.11 it didn't work before that

why do I need to add .cisco_imc folder? I added all the mibs files into the /mibs directory

**Comment by bastischubert** [comment]
@ranmalka 

> CISCO_FRU_URL used by 1.3.6.1.4.1.9.9.719.1.1.1.1.11 it didn't work before that

i checked, you only need those 4 mibs to get config working:
CISCO-UNIFIED-COMPUTING-FAULT-MIB.my
CISCO-UNIFIED-COMPUTING-MIB.my
CISCO-UNIFIED-COMPUTING-STORAGE-MIB.my
CISCO-UNIFIED-COMPUTING-TC-MIB.my

> why do I need to add .cisco_imc folder? I added all the mibs files into the /mibs directory
basically to prevent the download on every run of make mibs - plus putting that marker in a hidden file instead of a directory.

**Comment by ranmalka** [comment]
@bastischubert 

> 
> > CISCO_FRU_URL used by 1.3.6.1.4.1.9.9.719.1.1.1.1.11 it didn't work before that
> 
> i checked, you only need those 4 mibs to get config working: CISCO-UNIFIED-COMPUTING-FAULT-MIB.my CISCO-UNIFIED-COMPUTING-MIB.my CISCO-UNIFIED-COMPUTING-STORAGE-MIB.my CISCO-UNIFIED-COMPUTING-TC-MIB.my
> 

I also added cisco-TCI and CISCO-SMI  to your list, looks fine (tests passed)


>
> basically to prevent the download on every run of make mibs - plus putting that marker in a hidden file instead of a directory.

Now is that ok?

let me know what else is needed

**Review by bastischubert — APPROVED** [review]
LGTM

**Comment by bastischubert** [comment]
@ranmalka thanks for your patience and your contribution 🚀

**Pull request merged**